infinispan/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanEmbedded.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispan;
+
+import org.hibernate.ogm.cfg.OgmProperties;
+import org.hibernate.ogm.datastore.infinispan.options.navigation.InfinispanGlobalContext;
+import org.hibernate.ogm.datastore.infinispan.options.navigation.impl.InfinispanEntityContextImpl;
+import org.hibernate.ogm.datastore.infinispan.options.navigation.impl.InfinispanGlobalContextImpl;
+import org.hibernate.ogm.datastore.infinispan.options.navigation.impl.InfinispanPropertyContextImpl;
+import org.hibernate.ogm.datastore.spi.DatastoreConfiguration;
+import org.hibernate.ogm.options.navigation.spi.ConfigurationContext;
+
+/**
+ * Allows to configure options specific to the Infinispan data store.
+ */
+public class InfinispanEmbedded implements DatastoreConfiguration<InfinispanGlobalContext> {
```

Comment:
Good question. I assumed yes, especially as we want to keep the dependencies isolated from each other so that we can upgrade them independently.

Some options will necessarily be different as they are quite different; but indeed they will share some concepts. I'd stick to share names were it makes sense, but different API and package.